Explicit select SHA256 for code signing on windows

daschuer opened the pull request:
This fixes the following error message: 

```
SignTool Error: No file digest algorithm specified. Please specify the digest algorithm with the /fd flag. Using /fd SHA256 is recommended and more secure than SHA1. Calling signtool with /fd sha1 is equivalent to the previous behavior. In order to select the hash algorithm used in the signing certificate's signature, use the /fd certHash option. 
```
This restores the original behavior as a band aid. I have no clue what it means to switch to the recommended /fd SHA256

daschuer committed:
Select sha1 for code signing on windows

Swiftb0y commented:
IMO its just a change in hash strength. Changing it to the more secure sha256 should make no difference IMO. Do we hardcode the hashes somewhere else?

daschuer commented:
We can just give it a try. I will add a commit on top of this for SHA256 merge it and ask a windows user if anything it wrong. In case yes, quickly revert. Unfortunately we need to do this in main because if the secrets.

daschuer committed:
Use it SHA256 for code signing on Windows

Swiftb0y commented:
Does it really have to be on main? Can't we just quickly create a different branch based on main in the mixxx repo?

Swiftb0y commented:
https://github.com/mixxxdj/mixxx/tree/signtool_sha256

Swiftb0y commented on CMakeLists.txt:
```diff
@@ -1326,9 +1326,9 @@ if(WIN32)
     set(WINDOWS_CODESIGN_CERTIFICATE_PASSWORD "$ENV{WINDOWS_CODESIGN_CERTIFICATE_PASSWORD}" CACHE STRING "Password of signtool certificate")
     if("${WINDOWS_CODESIGN_CERTIFICATE_PATH}" STREQUAL "" AND "${WINDOWS_CODESIGN_CERTIFICATE_PASSWORD}" STREQUAL "")
-        set(WINDOWS_CODESIGN_ARGS /a /t http://timestamp.verisign.com/scripts/timstamp.dll CACHE STRING "parameters for signtool (list)")
+        set(WINDOWS_CODESIGN_ARGS /fd SHA256 /a /t http://timestamp.verisign.com/scripts/timstamp.dll CACHE STRING "parameters for signtool (list)")
```
we also need to specify the timestamp digest algorithm.
See the note at the top of the page: https://docs.microsoft.com/en-us/windows/win32/seccrypto/signtool

Swiftb0y replied:
I'm actually not sure about that anymore. We don't actually timestamp the build right now. I don't know what the purpose of that would be.

daschuer committed:
Set timstamp server explicit to http://timestamp.digicert.com

This is required according to https://docs.microsoft.com/en-us/windows/win32/seccrypto/signtool

daschuer committed:
Add missing parameter to windows code signing

/tr http://timestamp.digicert.com /fd SHA256

Swiftb0y commented:
You also need to set the timestamp digest algorithm
https://github.com/mixxxdj/mixxx/commit/d60036155301c24d68002f8bc3919bf589300a02
https://github.com/mixxxdj/mixxx/runs/7031560221

The build succeeded but it doesn't actually do any timestamping because we didn't specify `/t`.If we do specify the timestamp server, the build fails because the timestamp url is invalid?!?

daschuer commented:
Closed in favor of https://github.com/mixxxdj/mixxx/pull/4824
